File: polycarp_and_the_day_of_pi/poly.cpp
#include <iostream>
#include <cstdio>
#include <string>

using namespace std;
int main()
{
    int case_num;
    scanf("%d", &case_num);
    string pi = "314159265358979323846264338327";
    for (int i = 0; i < case_num; i++)
    {
        string word;
        int num_valid = 0;
        cin >> word;
        word.c_str();
        for (int j = 0; word[j] != '\0'; j++)
        {
            if (word[j] != pi[j])
            {
                break;
            }
            num_valid++;
        }

        printf("%d\n", num_valid);
    }
}

File: polycarp_and_the_day_of_pi/poly.py
def main():
    case_num = int(input())
    pi = "314159265358979323846264338327"
    for _ in range(case_num):
        digits = input()
        num_valid = 0
        for i in range(len(digits)):
            if (digits[i] != pi[i]):
                break
            num_valid += 1
        print(num_valid)
        
    
    
    
    
    
    
    
    
    
    
    
    
    
if __name__=="__main__":
    main()